### PR TITLE
virtctl: Allow to infer [instancetype|preference] when creating a VM

### PR DESCRIPTION
**Pull request by 0xFelix**
**What this PR does / why we need it**:

This adds the flags --infer-instancetype and --infer-preference.
When specified the instancetype and/or preference to be used are inferred
from the first volume of a created VM. The flags are mutually exclusive to
the --instancetype respectively --preference flag.

**Release note**:

```release-note
virtctl: Add options to infer instancetype and preference when creating a VM
```

### CODE REVIEW

File: pkg/virtctl/create/vm.go [hunk 1]
```diff
@@ -222,6 +237,12 @@ func (c *createVM) usage() string {
   # Create a manifest for a VirtualMachine with a cloned DataSource in namespace and specified size
   {{ProgramName}} create vm --volume-datasource=src:my-ns/my-ds,size:50Gi
 
+  # Create a manifest for a VirtualMachine with a cloned DataSource and an inferred instancetype
+  {{ProgramName}} create vm --volume-datasource=src:my-annotated-ds --infer-instancetype
+
+  # Create a manifest for a VirtualMachine with a cloned DataSource and an inferred preference
```
Comment (lyarwood) [inline pkg/virtctl/create/vm.go:243]: supernit - could just be one example here with both flags

File: pkg/virtctl/create/vm.go [hunk 2]
```diff
@@ -106,6 +112,8 @@ var optFns = map[string]optionFn{
 	CloudInitNetworkDataFlag: withCloudInitNetworkData,
 }
 
+// Until a param to control the boot order is introduced volumes have the following fixed priority:
+// Containerdisk < DataSource < Clone PVC < PVC
```
Comment (lyarwood) [inline pkg/virtctl/create/vm.go:116]: Can you add this ordering of `Volumes` to the user visible help for each flag?

**Review by lyarwood — CHANGES_REQUESTED** [review]
LGTM overall I just want to surface the ordering of `Volumes` in the user visible help text so they can understand why they always see a `DataSource` picked ahead of a `PVC` etc.

**Comment by lyarwood** [comment]
@davidvossel FYI we are trying to sneak support for `InferFromVolume` into `virtctl` for `v0.59.0`. Hopefully straight forward but your thoughts and review are welcome if you have time.

**Comment by 0xFelix** [comment]
@lyarwood Addressed both comments.

**Comment by lyarwood** [comment]
/lgtm

Thanks!

**Review by davidvossel — COMMENTED** [review]
/approve

**Pull request merged**